bread/middleware.py

```diff
@@ -7,6 +9,11 @@ def __init__(self, get_response):
         self.get_response = get_response
 
     def __call__(self, request):
+        if "bread-cookies" not in request.session:
+            request.session["bread-cookies"] = {}
+        for k, v in request.COOKIES.items():
+            if k.startswith("bread-"):
+                request.session["bread-cookies"][k] = urllib.parse.unquote(v)
```

Comment:
This is the middleware extension for makeing clients-side "bread" cookie available and persistent in the current session.